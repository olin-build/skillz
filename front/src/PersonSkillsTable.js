import React, { Component } from 'react';
import { graphql, withApollo } from 'react-apollo';

import { Rating } from './Rating.js'
import { getPersonSkillRecords } from './EditPerson.js'
import gql from 'graphql-tag'

export const PersonSkillsTable = ({ data, editablePerson = null, onRowClick }) => {
    if (data.error) {
        console.error(data.error.message)
        return (<div className="ui warning icon message">
            <i className="warning sign icon"></i>
            <div className="content">
                <div className="header"> Server Error </div>
                <p>{String(data.error.message)}</p>
                <p>If you are outside the Olin Network, turn on your VPN and try again.</p>
            </div>
        </div>)
    }
    if (data.loading) {
        return (<div className="ui active inverted dimmer">
            <div className="ui text loader">Loading</div>
        </div>)
    }
    let people = data.allPeople.edges.map(({ node }) => node);
    let skills = data.allSkills.edges.map(({ node }) => node);
    people.sort(({ firstName: a }, { firstName: b }) => a < b ? -1 : a > b ? 1 : 0)
    return (<table className="ui striped selectable definition table">
        <thead>
            <tr className="sticky">
                <th />
                {skills.map(({ name, id }) => <th key={id}>{name}</th>)}
            </tr>
        </thead>
        <tbody>
            {people.map(person =>
                <PersonSkillRow key={person.id}
                    person={person}
                    skills={skills}
                    editable={person === editablePerson}
                    onClick={() => onRowClick(person)}
                />)}
        </tbody>
    </table>)
}
export const PersonSkillRow = ({ person, skills, onClick, editable }) => {
    const personSkillRecords = getPersonSkillRecords(person, skills);
    return (<tr onClick={onClick}>
        <th className="person-name">
            <div className="first-name">{person.firstName}</div>
            <div className="right aligned last-name">{person.lastName}</div>
            {editable && <div className="ui label">
                Edit your stars in the table below.
      </div>}
        </th>
        {personSkillRecords.map(({ skill, personSkill }) =>
            <td key={skill.id} className="center aligned">
                <Rating rating={personSkill.experience} icon="star" />
                <Rating rating={personSkill.interest} icon="student" />
            </td>
        )}
    </tr >)
};

export const personSkillsQuery = gql`
query {
    allPeople {
        edges {
          node {
        id
          firstName
          lastName
          personSkillsByPersonId {
          edges {
        node {
          id
                skillId
        experience
                interest
                skillBySkillId {name}
        }
            }
          }
        }
      }
    }
    allSkills(orderBy: NAME_ASC) {
          edges {
        node {
          id
          name
        }
      }
    }
}`;


export const PersonSkillsTableContainer = graphql(personSkillsQuery, {
    options: {
        errorPolicy: 'all'
    }
})(PersonSkillsTable);
