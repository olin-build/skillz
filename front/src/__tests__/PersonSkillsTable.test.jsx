import React from 'react';
import renderer from 'react-test-renderer';
import { PersonSkillsTable } from '../PersonSkillsTable';

const personSkillsData = {
  allPeople: {
    edges: [
      {
        node: {
          id: 1,
          firstName: 'Laszlo',
          lastName: 'Toth',
          personSkillsByPersonId: {
            edges: [
              {
                node: {
                  id: 3,
                  skillId: 1,
                  experience: 5,
                  interest: 3,
                  note: null,
                  skillBySkillId: {
                    name: 'React',
                  },
                },
              },
            ],
          },
        },
      },
    ],
  },
  allSkills: {
    edges: [
      {
        node: {
          id: 13,
          name: 'React',
        },
      },
      {
        node: {
          id: 7,
          name: 'Jest',
        },
      },
    ],
  },
};

describe('PersonSkillsTable', () => {
  test('matches snapshot', () => {
    const component = renderer.create(<PersonSkillsTable data={personSkillsData} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
