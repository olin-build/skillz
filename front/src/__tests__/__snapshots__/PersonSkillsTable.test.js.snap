// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonSkillsTable matches snapshot 1`] = `
<table
  className="ui striped selectable definition table"
>
  <thead>
    <tr
      className="sticky"
    >
      <th />
      <th>
        React
      </th>
      <th>
        Jest
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      onClick={[Function]}
    >
      <th
        className="person-name"
      >
        <div
          className="first-name"
        >
          Laszlo
        </div>
        <div
          className="right aligned last-name"
        >
          Toth
        </div>
      </th>
      <td
        className="center aligned"
      >
        <div
          className="compact my-rating"
        >
          <i
            className="tiny empty star icon"
            style={
              Object {
                "color": "transparent",
              }
            }
          />
        </div>
        <div
          className="compact my-rating"
        >
          <i
            className="tiny empty star icon"
            style={
              Object {
                "color": "transparent",
              }
            }
          />
        </div>
      </td>
      <td
        className="center aligned"
      >
        <div
          className="compact my-rating"
        >
          <i
            className="tiny empty star icon"
            style={
              Object {
                "color": "transparent",
              }
            }
          />
        </div>
        <div
          className="compact my-rating"
        >
          <i
            className="tiny empty star icon"
            style={
              Object {
                "color": "transparent",
              }
            }
          />
        </div>
      </td>
    </tr>
  </tbody>
</table>
`;
